.github/workflows/_rust.yml

```diff
@@ -34,7 +34,9 @@ jobs:
           cargo doc --all-features --no-deps --document-private-items ${{ matrix.packages }}
         env:
           RUSTDOCFLAGS: "-D warnings"
-      - run: |
+      - env:
+          RUSTFLAGS: "-Wunused-crate-dependencies"
```

Comment:
You should be able to set this lint in `Cargo.toml` under the recently introduced `lints` table.